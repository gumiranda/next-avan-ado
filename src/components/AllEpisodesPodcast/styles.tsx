import styled from 'styled-components';

export const Wrapper = styled.section`
  padding-bottom: 2rem;
  table {
    width: 100%;
    th,
    td {
      padding: 0.75rem 1rem;
      border-bottom: 1px solid var(--gray-100);
    }
    th {
      color: var(--gray-200);
      text-transform: uppercase;
      font: 500 1rem Lexend, sans-serif;
      text-align: left;
    }
    td {
      font-size: 1.2rem;
      img {
        width: 6rem;
        height: 5rem;
        border-radius: 1rem;
      }
      a {
        color: var(--gray-800);
        font-family: Lexend, sans-serif;
        font-weight: 600;
        text-decoration: none;
        line-height: 2rem;
        font-size: 1rem;
        &:hover {
          text-decoration: underline;
        }
      }
      button {
        width: 2rem;
        height: 2rem;
        background: var(--white);
        border: 1px solid var(--gray-100);
        border-radius: 0.5rem;
        font-size: 0;
        transition: filter 0.2s;
        img {
          width: 1.75rem;
          height: 1.75rem;
        }
        &:hover {
          filter: brightness(0.95);
        }
      }
    }
  }
`;
