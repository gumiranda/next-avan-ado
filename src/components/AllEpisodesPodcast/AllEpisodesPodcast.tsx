import * as S from './styles';
import { Episode } from '@/domain/entities';
import Image from 'next/image';
import Link from 'next/link';
import { useContext } from 'react';
import { PlayerContext } from '@/domain/contexts';
interface IProps {
  allEpisodes: Episode[];
  title: string;
}
const AllEpisodesPodcast = ({
  allEpisodes = [],
  title = 'Todos os episódios',
  ...props
}: IProps) => {
  const { play } = useContext(PlayerContext);

  return (
    <S.Wrapper>
      <h2>{title}</h2>
      <table cellSpacing={0}>
        <thead>
          <tr>
            <th></th>
            <th>Podcast</th>
            <th>Integrantes</th>
            <th>Data</th>
            <th>Duração</th>
          </tr>
        </thead>
        <tbody>
          {allEpisodes.map((episode) => {
            return (
              <tr key={episode.id}>
                <td style={{ width: 72 }}>
                  <Image
                    objectFit="cover"
                    width={192}
                    height={192}
                    src={episode.thumbnail}
                    alt={episode.title}
                  />
                </td>
                <td>
                  <Link href={`/episodes/${episode.id}`}>
                    <a>{episode.title}</a>
                  </Link>
                </td>
                <td>{episode.members}</td>
                <td style={{ width: 100 }}>{episode.publishedAt}</td>
                <td>{episode.durationAsString}</td>
                <td>
                  <button type="button" onClick={() => play(episode)}>
                    <img src="/play-green.svg" alt="Tocar episódio" />
                  </button>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </S.Wrapper>
  );
};
export default AllEpisodesPodcast;
