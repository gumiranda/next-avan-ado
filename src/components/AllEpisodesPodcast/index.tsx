import AllEpisodesPodcast from './AllEpisodesPodcast';
export default AllEpisodesPodcast;
