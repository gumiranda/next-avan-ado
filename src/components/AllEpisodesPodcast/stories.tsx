import { Story, Meta } from '@storybook/react/types-6-0';

import AllEpisodesPodcast from './AllEpisodesPodcast';

export default {
  title: 'AllEpisodesPodcast',
  component: AllEpisodesPodcast
} as Meta;

export const Basic: Story = (args) => (
  <AllEpisodesPodcast title="Todos os episÃ³dios" allEpisodes={[]} {...args} />
);
Basic.args = {
  title: 'test'
};
