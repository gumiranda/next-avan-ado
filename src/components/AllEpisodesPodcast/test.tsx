import { render, screen } from '@testing-library/react';

import AllEpisodesPodcast from './AllEpisodesPodcast';

describe('<AllEpisodesPodcast />', () => {
  it('should render the heading', () => {
    const { container } = render(<AllEpisodesPodcast />);
    expect(
      screen.getByRole('heading', { name: /AllEpisodesPodcast/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
