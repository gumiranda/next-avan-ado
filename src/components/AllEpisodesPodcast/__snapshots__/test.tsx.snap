// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllEpisodesPodcast /> should render the heading 1`] = `
.c0 {
  padding-bottom: 2rem;
}

.c0 table {
  width: 100%;
}

.c0 table th,
.c0 table td {
  padding: 0.75rem 1rem;
  border-bottom: 1px solid var(--gray-100);
}

.c0 table th {
  color: var(--gray-200);
  text-transform: uppercase;
  font: 500 1rem Lexend,sans-serif;
  text-align: left;
}

.c0 table td {
  font-size: 1.2rem;
}

.c0 table td img {
  width: 6rem;
  height: 5rem;
  border-radius: 1rem;
}

.c0 table td a {
  color: var(--gray-800);
  font-family: Lexend,sans-serif;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  line-height: 2rem;
  font-size: 1rem;
}

.c0 table td a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 table td button {
  width: 2rem;
  height: 2rem;
  background: var(--white);
  border: 1px solid var(--gray-100);
  border-radius: 0.5rem;
  font-size: 0;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c0 table td button img {
  width: 1.75rem;
  height: 1.75rem;
}

.c0 table td button:hover {
  -webkit-filter: brightness(0.95);
  filter: brightness(0.95);
}

<section
  class="c0"
>
  <h2>
    AllEpisodesPodcast
  </h2>
  <table
    cellspacing="0"
  >
    <thead>
      <tr>
        <th />
        <th>
          Podcast
        </th>
        <th>
          Integrantes
        </th>
        <th>
          Data
        </th>
        <th>
          Duração
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
